### ISSUE TITLE
Token wasting, while debugging app deployment in Docker (or similar) 

### ISSUE THREAD

**Issue by m-check1B** [cline/cline#709]
### Which API Provider are you using?

Anthropic

### Which Model are you using?

Claude 3.5. Sonnet

### What happened?

Token wasting, while debugging docker deployment (or similar) Cline executes terminal command, gets 20 lines of command response, which is fine, but then pulls kilometres of logs and tries to process them. That does not help him and also ofter freezes and crashes it with resulting need to restart VM or VS code or at least Cline. The problem is he puls logs with not apparent reasonable limit or ignores the limit if that is set in his internals. Also this is by far the most expensive action in using the cline. By far. Like 10x. This seems to me as the only main and biggest problem left, after many recent improvements on of Cline.

### Steps to reproduce

1. lets Cline run some app in the Docker, but make wrong config
2. observe how cline waits for the terminal command response, you can even past it in to him while he waits.
3. now see he pulls all the long long long long long logs
4. he chuckles / or runs super expensive

### Relevant API REQUEST output

```shell
not needed
```

### Additional context

not needed

**Comment by scottmraz** [comment]
I usually just run commands like docker or NPM, then paste in errors. I see this as a feature enhancement more than a bug though, as it command output works as intended, when it works lol. Separate issue of it not using command output about 30% of the time.

**Comment by volome** [comment]
yes, it can be complicated but if you try with another model then Claude 3.5. Sonnet to do docker you will end up in space
